app/models/rpc/package.scala

```diff
@@ -146,6 +147,11 @@ package object rpc {
   implicit val StreamStateReads = enumReads(StreamState)
   implicit val DSAResponseFormat: Format[DSAResponse] = Json.format[DSAResponse]
 
+  implicit val QoSFormat: Format[models.akka.QoS.Level] = new Format[QoS.Level] {
+    override def reads(json: JsValue): JsResult[QoS.Level] = JsSuccess(QoS(json.as[Int]))
```

Comment:
`QoS.apply()` will fail for an invalid level value; need logic to produce either `JsSuccess` or `JsError`, probably the easiest way would be via `Try(...).fold(...)`

Reply:
removed all serializers for this. In next PR planning to replace QoS with Int